pkg/resourcemanager/apis/config/validation/validation.go

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validation
+
+import (
+	"time"
+
+	apivalidation "k8s.io/apimachinery/pkg/api/validation"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	componentbaseconfigvalidation "k8s.io/component-base/config/validation"
+	"k8s.io/utils/pointer"
+
+	"github.com/gardener/gardener/pkg/logger"
+	"github.com/gardener/gardener/pkg/resourcemanager/apis/config"
+)
+
+// ValidateResourceManagerConfiguration validates the given `ResourceManagerConfiguration`.
+func ValidateResourceManagerConfiguration(conf *config.ResourceManagerConfiguration) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, validateSourceClientConnection(conf.SourceClientConnection, field.NewPath("sourceClientConnection"))...)
+	if conf.TargetClientConnection != nil {
+		allErrs = append(allErrs, validateTargetClientConnection(*conf.TargetClientConnection, field.NewPath("targetClientConnection"))...)
+	}
+	allErrs = append(allErrs, validateServerConfiguration(conf.Server, field.NewPath("server"))...)
+	allErrs = append(allErrs, componentbaseconfigvalidation.ValidateLeaderElectionConfiguration(&conf.LeaderElection, field.NewPath("leaderElection"))...)
+
+	if conf.LogLevel != "" && !sets.NewString(logger.AllLogLevels...).Has(conf.LogLevel) {
```

Comment:
An empty LogLevel should not be allowed
```suggestion
	if !sets.NewString(logger.AllLogLevels...).Has(conf.LogLevel) {
```